opentelemetry-aws/src/lib.rs

```diff
@@ -86,6 +86,107 @@ pub mod trace {
         _private: (),
     }
 
+    pub fn deserialize_span_context(value: &str) -> Result<SpanContext, ()> {
+        let parts: Vec<(&str, &str)> = value
+            .split_terminator(';')
+            .filter_map(from_key_value_pair)
+            .collect();
+
+        let mut trace_id: TraceId = TraceId::INVALID;
+        let mut parent_segment_id: SpanId = SpanId::INVALID;
+        let mut sampling_decision = TRACE_FLAG_DEFERRED;
+        let mut kv_vec: Vec<(String, String)> = Vec::with_capacity(parts.len());
+
+        for (key, value) in parts {
+            match key {
+                HEADER_ROOT_KEY => {
+                    let converted_trace_id: Result<TraceId, ()> =
+                        XrayTraceId(value.to_string()).try_into();
+                    match converted_trace_id {
+                        Err(_) => return Err(()),
+                        Ok(parsed) => trace_id = parsed,
+                    }
```

Comment:
```suggestion
                    match TraceId::from(XrayTraceId(value.to_string()) {
                        Err(_) => return Err(()),
                        Ok(parsed) => trace_id = parsed,
                    }
```

Also, it's a bit silly/sad that we currently require the allocation (in `value.to_string()`) in order to parse from `XrayTraceId` to `TraceId`. We should either extract that `TryFrom` impl into a separate method or potentially change `XrayTraceId`'s inner value to be a `Cow<'a, str>`.

Reply:
Fixed. And refactored `XrayTraceId` to avoid memory allocation